File: easy/0696_count_binary_substrings/solution.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        count = 0

        prev = 0
        curr = 1

        for i in range(len(s) - 1):
            if s[i] == s[i + 1]:
                curr += 1
            else:
                count += min(curr, prev)
                prev = curr
                curr = 1

        return count + min(curr, prev)

File: easy/0696_count_binary_substrings/solution.cpp
#include <string>

class Solution {
public:
  int countBinarySubstrings(std::string s) {
    int count = 0;

    int prev = 0;
    int curr = 1;

    for (size_t i = 1; i < s.size(); ++i) {
      if (s[i - 1] == s[i])
        ++curr;
      else {
        count += std::min(curr, prev);
        prev = curr;
        curr = 1;
      }
    }

    return count + std::min(curr, prev);
  }
};
